docs/core/entity/todo.md

```diff
@@ -0,0 +1,109 @@
+| `UPDATE_TODO_ITEM`      | Entity implements the methods to allow update of to-do items.  |
+| `MOVE_TODO_ITEM`        | Entity implements the methods to re-order to-do items.  |
+
+## Methods
+
+
+### Create To-do items
+
+A To-do list entity may support creating To-do items by specifying the `CREATE_TODO_ITEM`
+supported feature. The `TodoItem` field `uid` will not be present since it is a
+unique identifier determined by the integration when the item is created.
+
+```python
+from homeassistant.components.todo import TodoListEntity
+
+class MyTodoListEntity(TodoListEntity):
+
+    async def async_create_todo_item(self, item: TodoItem) -> None:
+        """Add an item to the To-do list."""
+```
+
+### Delete To-do items
+
+A To-do list entity may support deleting To-do items by specifying the `DELETE_TODO_ITEM`
+supported feature. Integrations must support deleting multiple items.
+
+```python
+from homeassistant.components.todo import TodoListEntity
+
+class MyTodoListEntity(TodoListEntity):
+
+    async def async_delete_todo_items(self, uids: list[str]) -> None:
+        """Add an item to the To-do list."""
+```
+
+### Update To-do items
+
+A To-do list entity may support updating To-do items by specifying the `UPDATE_TODO_ITEM`
+supported feature. The `TodoItem` field `uid` is always present and indicates
+which item should be updated, and all other fields are optional. Integrations
+must support partial update.
+
+```python
+from homeassistant.components.todo import TodoListEntity
+
+class MyTodoListEntity(TodoListEntity):
+
+    async def async_update_todo_item(self, item: TodoItem) -> None:
+        """Add an item to the To-do list."""
+```
+
+### Move To-do items
+
+A To-do list entity may support re-ordering To-do items in the list by specifying
+the `MOVE_TODO_ITEM` supported feature. The To-do item with the specified `uid`
+should be moved to the position in the list specified by `pos` (`0` is the first
+position in the To-do list).
+
+```python
+from homeassistant.components.todo import TodoListEntity
+
+class MyTodoListEntity(TodoListEntity):
+
+    async def async_move_todo_item(self, uid: str, pos: int) -> None:
+        """Move an item in the To-do list."""
+```
+
+## TodoItem
+
+A `TodoItem` represents an individual item on a To-do list. The methods
+above describe any differences about which fields are optional on create or
+update.
+
+| Name        | Type             | Default      | Description                                                                                                                                     |
+| ----------- | ---------------- | ------------ | ----------------------------------------------------------------------------------------------------------------------------------------------- |
+| uid | <code>string &#124; None</code> | `None` | **Required.** A unique identifier for the to-do item. This field is required for mutations.
```

Comment:
Why are these marked as required if the value is optional?

Reply:
Bump.

Reply:
Thanks for the bump, i thought i replied.

The thinking here was to consider this from the implementers perspective when returning the `TodoItem` used for state, where these fields are expected to be set. They are optional in the dataclass since the dataclass is used in create or update operations which have different rules. The comment *The methods
above describe any differences about which fields are optional on create or
update.* was meant to explain that.

An alternative is to mark these as all option, then re-explain what is required in the `todo_items` properties instructions above.

Happy to adjust based on your feedback.

Reply:
My last comment was ambiguous: Let me know what you think and i'll adjust.

Reply:
I would remove the "**Required.**" and instead have a sentence at the end for each field in the table that briefly describes when the field is required. We already have this for `uid` here.